src/commands/image/imageSource/containerRegistry/getRegistryCredentialsAndSecrets.ts

```diff
@@ -40,7 +40,7 @@ export async function getAcrCredentialsAndSecrets(context: IContainerRegistryIma
 export function getThirdPartyCredentialsAndSecrets(context: IContainerRegistryImageContext, containerAppSettings?: RegistryCredentialsAndSecrets): RegistryCredentialsAndSecrets {
     // If 'docker.io', convert to 'index.docker.io', else use registryName as loginServer
     const loginServer: string = (context.registryDomain === dockerHubDomain) ? dockerHubRegistry : nonNullProp(context, 'registryName').toLowerCase();
-    const passwordSecretRef: string = `${loginServer.replace(/[\.]+/g, '')}-${context.username}`;
+    const passwordSecretRef: string = `${loginServer.replace(/[^a-z0-9-]+/g, '')}-${context.username}`;
```

Comment:
I think based on the error message this should also include a "." character.

Reply:
This is actually intentional, otherwise this will happen:

![image](https://github.com/microsoft/vscode-azurecontainerapps/assets/40250218/f1b22ff8-febf-4629-a955-e5343803a183)

I'm not really sure why we're seeing a different error than the previously posted one but we can avoid both errors pretty easily by just omitting "."